IPython/core/tests/test_magic.py

```diff
@@ -1121,6 +1125,36 @@ def test_script_config():
     assert "whoda" in sm.magics["cell"]
 
 
+def _interrupt_after_1s():
+    sleep(1)
+    signal.raise_signal(signal.SIGINT)
+
+
+def test_script_raise_on_interrupt():
+    ip = get_ipython()
+
+    with pytest.raises(CalledProcessError):
+        thread = Thread(target=_interrupt_after_1s)
+        thread.start()
+        ip.run_cell_magic(
+            "script", f"{sys.executable}", "from time import sleep; sleep(2)"
```

Comment:
```suggestion
            "script", f"{sys.executable}", "from time import sleep; sleep(2)"
```

Do you want a `-c` in here ? Otherwise you might be just starting the cpython repl?

Reply:
I think this is fine as this is the same pattern as used in all other `%%script` tests used in this file

Reply:
My guess is something like that triggers the pytest warning

Reply:
So it turns out it that this is down to `test_debug_magic` and `test_debug_magic_locals`. If these are commented out then the error does not show up when running the file locally. Double checking on CI to see if there is more tests like that. It seems like the teardown for these two tests is incomplete.

Reply:
Yup, back to green on CI. It looks like an interaction between `pytest` and `%debug`.